kickstarter/android-oss
# NT-1880: Button still active after signing in

# 📲 What

Button still active after signing in

# 🤔 Why

Sign in button remains active after being pressed

# 🛠 How

- Added observer to button clicked observable

```

logInButtonClicked
                .compose(bindToLifecycle())
                .subscribe {
                    this.lake.trackLogInButtonCtaClicked()
                    this.logInButtonIsEnabled.onNext(false)
                }

```

- Renabled button on netword respose 
```
   emailAndPassword
                .compose(takeWhen(this.logInButtonClicked))
                .compose(bindToLifecycle())
                .switchMap { ep -> this.client.login(ep.first, ep.second).materialize() }
                .share()
                .subscribe {
                    this.logInButtonIsEnabled.onNext(true)
                    errors.onNext(unwrapNotificationEnvelopeError(it))
                    successResponseData.onNext(unwrapNotificationEnvelopeSuccess(it))
                }

```


# 👀 See

Trello, screenshots, external resources?

| Before 🐛 | After 🦋 |
| --- | --- |
|  |  |

# 📋 QA

- Go to the ``` Login Screen ```
- Fill in the required fields and click on the ``` Log in ``` button.

The ``` Log in ``` button should be disabled as expected.

# Story 📖

https://kickstarter.atlassian.net/browse/NT-1880
